src/polyswarm_api/__main__.py

```diff
@@ -227,28 +246,24 @@ def search():
 
 
 @click.option('-r', '--hash-file', help='File of hashes, one per line.', type=click.File('r'))
-@click.option('--hash-type', help='Hash type to search [sha256|sha1|md5], default=sha256', default='sha256')
 @click.argument('hashes', nargs=-1)
 @search.command('hash', short_help='search for hashes separated by space')
 @click.pass_context
-def hashes(ctx, hashes, hash_file, hash_type):
+def hashes(ctx, hashes, hash_file):
     """
     Search PolySwarm for files matching sha256 hashes
     """
 
     def _get_hashes_from_file(file):
         return [h.strip() for h in file.readlines()]
 
-    def _remove_invalid_hashes(hash_candidates, candidates_hash_type):
-
-        def is_valid_hash(hash_candidate):
-            return (candidates_hash_type == 'sha256' and _is_valid_sha256(hash_candidate)) or \
-                   (candidates_hash_type == 'sha1' and _is_valid_sha1(hash_candidate)) or \
-                   (candidates_hash_type == 'md5' and _is_valid_md5(hash_candidate))
-
+    # filter before API call so it does not raise exception
+    # and break execution
+    def _remove_invalid_hashes(hash_candidates):
```

Comment:
we want to force the user to specify the type of hash. And if they do not specify the type of hash then we will assume sha256. So, I need you to re-add the "hash_type" arg to all of these functions. Then update the docs to tell the user to specify the hash type on the command line. And if no hash type is provided, then we will assume sha256.
Based on the type of hash they provide, we want to validate it to ensure the hash provided matches the hash type specified.